### PR TITLE
Fix bugs in absolute urls

### PR DESCRIPTION
**Pull request by orangejenny**
##### SUMMARY
Introduced in https://github.com/dimagi/commcare-hq/pull/26935/

##### PRODUCT DESCRIPTION
Fixes 500s on user registration.

### CODE REVIEW

**Review by calellowitz — APPROVED** [review]

File: corehq/apps/registration/tasks.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def activation_24hr_reminder_email():
 
 @task(serializer='pickle', queue="email_queue")
 def send_domain_registration_email(recipient, domain_name, guid, full_name, first_name):
-    registration_link = 'http://' + DNS_name + reverse('registration_confirm_domain') + guid + '/'
+    registration_link = 'http://' + get_site_domain() + reverse('registration_confirm_domain') + guid + '/'
```
Comment (biyeun) [inline corehq/apps/registration/tasks.py]: @orangejenny i think there might be a problem here if the `http://` is left in?

**Review by biyeun — CHANGES_REQUESTED** [review]

**Review by biyeun — APPROVED** [review]

File: corehq/apps/registration/tasks.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/registration/tasks.py]: What do you mean? All I'm intending to do is restore the old code: https://github.com/dimagi/commcare-hq/pull/26935/files#diff-0890a99a4eb7f4a7eace79122bf9574cL62-L63

Comment (biyeun) [reply corehq/apps/registration/tasks.py]: @orangejenny i read the commit incorrectly...all good

**Review by millerdev — APPROVED** [review]

**Comment by orangejenny** [comment]
Build has passed, GitHub just isn't reflecting that yet.

**Comment by czue** [comment]
:disappointed: sorry about that - thanks for handling

**Pull request merged**